wordpress-mobile/WordPress-Android
# Adds tracking coverage to the My Sites No Sites view

## Description
This adds tracking coverage to the My Sites: No Sites view when:

1. The view is displayed: `my_site_no_sites_view_displayed`
2. The action button is tapped: `my_site_no_sites_view_action_tapped`
3. The view is hidden: `my_site_no_sites_hidden`

## To test:

1. Launch the app with an account that you have sites linked to
2. Verify the following events are not logged: `my_site_no_sites_view_displayed`, `my_site_no_sites_view_action_tapped`, `my_site_no_sites_hidden`
3. Logout of that account, and sign into an account that has no sites link it (or create a new account and skip the site creation process)
4. After logging in, verify you see `🔵 Tracked: my_site_no_sites_view_displayed <>` in the console
5. Tap the 'Add new site' button
6. Verify you see `🔵 Tracked: my_site_no_sites_view_action_tapped <>` in the console
7. Dismiss the site creation flow
8. Restart the app
9. Verify you see the following tracks in this order:
```
🔵 Tracked: application_opened <>
🔵 Tracked: my_site_tab_accessed <>
🔵 Tracked: my_site_no_sites_view_displayed <>
```
11. Tap the add site button again
7. Go through the site creation process 
8. After successfully creating your site verify you see `🔵 Tracked: my_site_no_sites_view_hidden <>` logged once
9. Delete the site using Site Settings
11. Restart the app
10. Verify `🔵 Tracked: my_site_no_sites_view_displayed <>` is logged in console

## Regression Notes
1. Potential unintended areas of impact
This only adds tracking and does not modify existing behavior. Therefore there should not be any unintended areas of impact. 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
None.

## PR submission checklist:
